benchmark/distributed/master.js

```diff
@@ -0,0 +1,163 @@
+'use strict';
+
+const yargs = require('yargs');
+const fs = require('fs');
+const path = require('path');
+
+const jstp = require('../..');
+const parseAddress = require('./parse-address');
+const statistics = require('../statistics');
+
+const args = yargs
+  .option('address', {
+    alias: 'a',
+    describe: 'Address for master server to listen on',
+    type: 'string',
+    default: 'tcp://localhost:3000',
+  })
+  .options('report', {
+    alias: 'r',
+    describe: 'Report interval in ms',
+    type: 'number',
+    default: 10000,
+  })
+  .options('key', {
+    alias: 'k',
+    describe: 'Path to file that contains SSL certificate key',
+    type: 'string',
+    coerce:
+      arg => (arg ? fs.readFileSync(path.resolve(__dirname, arg)) : undefined),
+  })
+  .options('cert', {
+    alias: 'c',
+    describe: 'Path to file that contains SSL certificate',
+    type: 'string',
+    coerce:
+      arg => (arg ? fs.readFileSync(path.resolve(__dirname, arg)) : undefined),
+  })
+  .help()
+  .alias('help', 'h')
+  .argv;
+
+const {
+  address,
+  report: reportInterval,
+  key,
+  cert,
+} = args;
+
+const { transport, listen: listenAddress } = parseAddress(address);
+
+let startTimeHR = null;
+let workersReportBuffer = [];
+const workersReports = [];
+const serverReports = [];
+
+const workers = [];
+let server = null;
+let master = null;
+
+let serverAddress = null;
+
+const application = {
+  master: {
+    registerServer: (connection, address, callback) => {
+      if (server) {
+        callback('Server is already registered');
+      }
+      server = connection;
+      serverAddress = address;
+      workers.forEach((worker) => {
+        worker.emitRemoteEvent('worker', 'connect', [serverAddress]);
+      });
+      connection.on('event', (iface, event, args) => {
+        if (iface !== 'master') {
+          return;
+        }
+        if (event === 'report') {
+          const count = +args[0];
+          serverReports.push(args);
+          console.log(`Server counnections: ${count}`);
+        }
+      });
+      callback(null);
+    },
+
+    registerWorker: (connection, callback) => {
+      workers.push(connection);
+      if (serverAddress) {
+        connection.emitRemoteEvent('master', 'connect', [serverAddress]);
+      }
+
+      connection.on('event', (iface, event, args) => {
+        if (iface !== 'master') {
+          return;
+        }
+        if (event === 'report') {
+          const requests = +args[0];
+          workersReportBuffer.push(requests);
```

Comment:
Yep, this way in the future, we can save the data to a file, and then analyze it with some tool. We can't analyze RPS's because that value is not precise.